### ISSUE TITLE
Creation of archivematica user in postinstall scripts for CentOS 7 needs more robust checks

### ISSUE THREAD

**Issue by jpellman** [artefactual/archivematica#955]
During an attempt at installing Archivematica 1.7 RC on a fresh CentOS 7 host, I noticed the following line pop up while I was installing _archivematica-storage-service_:

```
creating archivematica user
id: ‘archivematica’: no such user
Adding system user `archivematica' (UID 333) ...
```
The _no such user_ error piqued my curiosity so I checked what the pre-/post-install scripts were doing.

```
[centos@ip-172-31-17-204 tmp]$  sudo rpm -q --scripts archivematica-storage-service
postinstall scriptlet (using /bin/sh):

# Create archivematica user
userID=`id -u archivematica`
if [ "${userID}" != 333 ]; then
  useradd --uid 333 --user-group --home /var/lib/archivematica/ archivematica
fi
```

As-is, this user creation code is problematic for a variety of reasons.  For instance, if there is already an Archivematica account with username _archivematica_ and a UID other than 333 (say, 1001), this will create another _archivematica_ account with UID 333. If the _archivematica_ with 1001 has an entry in _/etc/shadow_ already with a hashed password, this same password should also work with the _archivematica_ with UID 333, since the join between _/etc/passwd_ and _/etc/shadow_ is based on username rather than UID. This means that both accounts named _archivematica_ could be logged into with the same password, which seems like a scenario we wouldn't want (not to mention that if either _archivematica_ account changed their password, it would also be changed for the other account as well). Also, this code doesn't check whether or not there is already a user out there with UID 333 on the host before running useradd.

The Fedora project has some suggestions for [better ways to do this](http://fedoraproject.org/wiki/Packaging%3aUsersAndGroups#Soft_static_allocation), but at a minimum the user creation code should ideally a) check that a user with UID 333 doesn't already exist and b) ensure that only one user account exists with the name _archivematica_.

(Leaving this untagged since it's out of scope for Phase 1.)

**Comment by sevein** [comment]
One more PR needs to be merged before this can considered fixed!
https://github.com/artefactual-labs/am-packbuild/pull/161

**Comment by sevein** [comment]
https://github.com/artefactual-labs/am-packbuild/pull/161 merged.
